src/Symfony/Component/HtmlSanitizer/Parser/ParserInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer\Parser;
+
+/**
+ * A parser transforms a user HTML input string into a tree of DOMNode objects.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @experimental
+ */
+interface ParserInterface
+{
+    /**
+     * Parse a given string and returns a DOMNode tree.
+     * This method must return null if the string cannot be parsed as HTML.
```

Comment:
How does it account for error handling?

Reply:
See https://github.com/symfony/symfony/pull/44681#discussion_r771536972

Reply:
the masterminds parser collects errors, though. Not sure hiding them is wise.

Reply:
It's now possible to inject a custom parser to gather errors if one wishes.